Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/cache/delta_fifo.go

```diff
@@ -154,10 +156,30 @@ func (f *DeltaFIFO) Update(obj interface{}) error {
 	return f.queueActionLocked(Updated, obj)
 }
 
-// Delete is just like Add, but makes an Deleted Delta.
+// Delete is just like Add, but makes an Deleted Delta. If the item does not
+// already exist, it will be ignored. (It may have already been deleted by a
+// Replace (re-list), for example.
 func (f *DeltaFIFO) Delete(obj interface{}) error {
+	id, err := f.KeyOf(obj)
+	if err != nil {
+		return KeyError{obj, err}
+	}
 	f.lock.Lock()
 	defer f.lock.Unlock()
+	if f.knownObjects == nil {
+		if _, exists := f.items[id]; !exists {
+			// Presumably, this was deleted when a relist happened.
+			// Don't provide a second report of the same deletion.
+			return nil
```

Comment:
I already submitted a pull to David's branch that fixes it. But I don't
think David is sold on my solution.

Ben Parees | OpenShift
On Oct 29, 2015 8:15 AM, "Jordan Liggitt" <notifications@github.com> wrote:

> In Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/cache/delta_fifo.go
> <https://github.com/openshift/origin/pull/5339#discussion_r43377301>:
>
> >  	f.lock.Lock()
> >  	defer f.lock.Unlock()
> > +	if f.knownObjects == nil {
> > +		if _, exists := f.items[id]; !exists {
> > +			// Presumably, this was deleted when a relist happened.
> > +			// Don't provide a second report of the same deletion.
> > +			return nil
>
> how much work is required to get the build controller working with this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5339/files#r43377301>.
>